Userland/Utilities/shot.cpp

```diff
@@ -5,28 +5,100 @@
  */
 
 #include <AK/Format.h>
+#include <AK/Optional.h>
 #include <LibCore/ArgsParser.h>
 #include <LibCore/DateTime.h>
 #include <LibCore/File.h>
 #include <LibGUI/Application.h>
 #include <LibGUI/Clipboard.h>
+#include <LibGUI/Painter.h>
+#include <LibGUI/Widget.h>
+#include <LibGUI/Window.h>
 #include <LibGUI/WindowServerConnection.h>
 #include <LibGfx/PNGWriter.h>
 #include <unistd.h>
 
+class SelectableLayover final : public GUI::Widget {
+    C_OBJECT(SelectableLayover)
+public:
+    SelectableLayover(GUI::Window* window)
+        : m_window(window) {};
+    virtual ~SelectableLayover() override {};
+    Optional<Gfx::IntRect> region() const
+    {
+        return m_region;
+    }
+
+private:
+    virtual void mousedown_event(GUI::MouseEvent& event) override
+    {
+        if (event.button() != GUI::MouseButton::Left)
+            return;
+
+        if (m_achor_point.has_value())
+            m_achor_point.clear();
+        else
+            m_achor_point = event.position();
+    };
+
+    virtual void mousemove_event(GUI::MouseEvent& event) override
+    {
+        m_last_point = event.position();
+        if (m_achor_point.has_value())
+            update();
+    };
+
+    virtual void paint_event(GUI::PaintEvent&) override
+    {
+        m_region = build_rect(m_achor_point, m_last_point);
+        GUI::Painter painter(*this);
+        painter.draw_rect(m_region.value_or(Gfx::IntRect()), Gfx::Color::Black);
+    }
+
+    virtual void keydown_event(GUI::KeyEvent& event) override
+    {
+        if (event.key() != Key_Return)
+            return;
+
+        m_window->close();
+    }
+
+    static Optional<Gfx::IntRect> build_rect(Optional<Gfx::IntPoint> const& p1, Optional<Gfx::IntPoint> const& p2)
+    {
+        if (!p1.has_value() || !p1.has_value())
```

Comment:
You probably want:
` if (!p1.has_value() || !p2.has_value())`

Reply:
Good catch :) but not necessary anymore as @MaxWipfli pointed to `Gfx::IntRect::from_two_points`.